### DISCUSSION TITLE
Does Jans support multiple databases - split configuration and identity data for example

### DISCUSSION THREAD

**Discussion by johnlister** [JanssenProject/jans#8993] [Q&A]
As the subject implies. Is it possible to use LDAP as the persistence mechanism for identity information and another database (PostgreSQL for example) for configuration (or session storage, etc)?

It appears all entities are stored in the same "database".

**Comment by ossdhaval** [comment]
Hi @johnlister 

I believe this is not possible. I am tagging @yuriyz for confirmation. 

As a side note, if you have an external identity store (LDAP/KeyCloak), you can use [Jans Link](https://docs.jans.io/v1.1.3/admin/link/) to connect them to your Janssen Server instance and continue to use them as your identity store.

**Comment by yuriyz** [comment]
It is better to ask @yurem jans-orm related question.

**Reply by johnlister** [reply]
Thanks both, our current setup - potentially looking at replacing with Jans - uses 2 ldap datasources (one for users - read optimised and one for session data - write optimised) along with a separate config datasource. I'm assuming that the mixed use case performance is one reason that Jans doesn't recommend ldap for production systems and why we have differently configured ldap sources for each type of data.

**Reply by nynymike** [reply]
We don't like LDAP anymore for prod deployments--it's ok for testing, but there are no cloud hosted LDAP offerings (e.g. Google Spanner, AWS Aurora).  Also, as both user data and configuration data have the same security concerns, I don't see a security advantage to split them apart. In fact, most of the outages are related to database. So adding complextity to the persistence layer doesn't make sense.  Is there any reason other then performance you want to do this?

**Reply by ossdhaval** [reply]
Tagging @johnlister

**Reply by johnlister** [reply]
Thanks @nynymike  we have our own production hardened LDAP clusters. For performance reasons they are split into multiple clusters, each configured for different use cases/load. In an ideal world we could leave them as is if we migrated to Jans.

**Reply by nynymike** [reply]
And your hardened LDAP replication clusters are better then RDBMS alternatives? Does't your organization also have an RDBMS service?

**Reply by johnlister** [reply]
We have maintained the LDAP clusters over many years, they are multi master write capable and can be scaled up or down easily. So yes, I'd argue that they are as good if not better than an RDBMS for this use case. Modern cloud based RDBMS may offer some benefits, but whilst cloud hosted DBs, etc work for many smaller organisations there are multiple caveats that apply where managing your own can be beneficial especially to large organisations.
We could use a cloud RDBMS (and we do use them for other use-cases) but we would have to migrate all of our data across whilst also keeping the service up 24/7 during the migration/switchover.

**Reply by nynymike** [reply]
> As the subject implies. Is it possible to use LDAP as the persistence mechanism for identity information and another database (PostgreSQL for example) for configuration (or session storage, etc)? 

No.

**Comment by nynymike** [comment]
![image](https://github.com/user-attachments/assets/ad40cecb-5c0f-48b9-9c88-0e02a7e27e7e)

**Reply by nynymike** [reply]
We are patching this version of OpenDJ, but it seems to always be the problem child. Also, we are not LDAP database programmers. Sometimes we add new algorithms for password hashing... but we don't do much other then patch.  Also deploying LDAP in Kubernetes is too much work.

**Comment by nynymike** [comment]
The answer is no.
